django/core/management/commands/dumpdata.py

```diff
@@ -6,6 +8,38 @@
 from django.core.management.utils import parse_apps_and_model_labels
 from django.db import DEFAULT_DB_ALIAS, router
 
+try:
+    import bz2
+    HAS_BZ2 = True
+except ImportError:
+    HAS_BZ2 = False
+
+try:
+    import lzma
+    HAS_LZMA = True
+except ImportError:
+    HAS_LZMA = False
+
+
+def get_open_method(filename, mode='wt'):
+    """Return open method an mode based on the filename suffix."""
+    try:
+        filename_suffix = pathlib.PurePath(filename).suffix
+    except TypeError:
+        filename_suffix = ''
+    compression_formats = {
+        '.gz': (gzip.open, mode),
+    }
+    if HAS_BZ2:
+        compression_formats['.bz2'] = (bz2.open, mode)
+    if HAS_LZMA:
+        compression_formats['.lzma'] = (lzma.open, mode)
+        compression_formats['.xz'] = (lzma.open, mode)
+    try:
+        return compression_formats[filename_suffix]
+    except KeyError:
+        return (open, mode)
```

Comment:
If we don't have BZ2 or LZMA, we should fail to save in those formats rather than fall back to uncompressed. It's just confusing when files are uncompressed when they have an extension implying they are.

Reply:
The current behavior is to save in plain text even if a compressed file extension is specified so I kept this behavior.

Reply:
But as I said, I think that is a bad idea as it will have the wrong extension on it. The two acceptable choices to deal with this are:

1. Fail loudly as the user expected a compressed file.
2. Strip the extension and continue with an uncompressed file, but show a warning.

Reply:
I implemented (2)